packages/client/src/events/open-feature-event-emitter.ts

```diff
@@ -1,14 +1,14 @@
 import { GenericEventEmitter } from '@openfeature/core';
 import EventEmitter from 'events';
-
+import { ProviderEmittableEvents } from './events';
 /**
  * The OpenFeatureEventEmitter can be used by provider developers to emit
  * events at various parts of the provider lifecycle.
  *
  * NOTE: Ready and error events are automatically emitted by the SDK based on
  * the result of the initialize method.
  */
-export class OpenFeatureEventEmitter extends GenericEventEmitter {
+export class OpenFeatureEventEmitter extends GenericEventEmitter<ProviderEmittableEvents> {
```

Comment:
This restricts providers from manually emitting `PROVIDER_CONTEXT_CHANGED` (we only want the SDK to do that).